### ISSUE TITLE
Problems with dependency not being found by meson

### ISSUE THREAD

**Issue by ChukwuemezieChristopherAneke** [mesonbuild/meson#3920]
I am compiling a pdf viewer which needs synctex for some additional features. When I run the `meson` command it does not find this file. Adding the root of the synctex executable to library path like this

`export LIBRARY_PATH=usr/local/texlive/2018/bin/x86_64-linux/`

does not work. On my system all the locations of this file are:

```
usr/local/texlive/2018/tlpkg/tlpobj/synctex.x86_64-linux.tlpobj
/usr/local/texlive/2018/tlpkg/tlpobj/synctex.tlpobj
/usr/local/texlive/2018/texmf-dist/doc/man/man5/synctex.5
/usr/local/texlive/2018/texmf-dist/doc/man/man5/synctex.man5.pdf
/usr/local/texlive/2018/texmf-dist/doc/man/man1/synctex.man1.pdf
/usr/local/texlive/2018/texmf-dist/doc/man/man1/synctex.1
/usr/local/texlive/2018/texmf-dist/doc/context/sources/general/manuals/workflows/workflows-synctex.tex
/usr/local/texlive/2018/texmf-dist/tex/context/interface/mkiv/i-synctex.xml
/usr/local/texlive/2018/texmf-dist/scripts/context/lua/mtx-synctex.lua
/usr/local/texlive/2018/bin/x86_64-linux/synctex
/usr/lib64/libsynctex.so.1
/usr/lib64/libsynctex.so.1.18.0


```
How do I get meson to recognise and find it?

**Comment by jpakkane** [comment]
`LIBRARY_PATH` does not mean anything (and even if it did, it would probably need to be an absolute path). What file are you trying to find and how? Please post the full log and build file. Otherwise this can't really be debugged.

**Comment by ChukwuemezieChristopherAneke** [comment]
I'm trying to find synctex, the reference for `LIBRARY_PATH` was [this issue](https://github.com/mesonbuild/meson/issues/217 "meson github issue"). I'm still not as familiar with meson as I am with `cmake and make`. Which build file are you referring to? 

The full log is:

```
Build started at 2018-07-20T17:53:45.298163
Main binary: /usr/bin/python3
Python system: Linux
The Meson build system
Version: 0.46.1
Source dir: /home/user/zathura/src
Build dir: /home/user/zathura/src/build
Build type: native build
Project name: zathura
Sanity testing C compiler: ccache cc
Is cross compiler: False.
Sanity check compiler command line: ccache cc /home/user/zathura/src/build/meson-private/sanitycheckc.c -o /home/user/zathura/src/build/meson-private/sanitycheckc.exe
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /home/user/zathura/src/build/meson-private/sanitycheckc.exe
Native C compiler: ccache cc (gcc 8.1.1 "cc (GCC) 8.1.1 20180502 (Red Hat 8.1.1-1)")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Running compile:
Working directory:  /tmp/tmpds6avq7w
Command line:  ccache cc /tmp/tmpds6avq7w/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -o /tmp/tmpds6avq7w/output.exe -O0 -Wl,--start-group -llibm -Wl,--end-group 

Code:
 int main(int argc, char **argv) { return 0; }
Compiler stdout:
 
Compiler stderr:
 /usr/bin/ld: cannot find -llibm
collect2: error: ld returned 1 exit status

Library libm found: YES
Found pkg-config: /usr/bin/pkg-config (1.4.2)
Determining dependency 'girara-gtk3' with pkg-config executable '/usr/bin/pkg-config'
Native dependency girara-gtk3 found: YES 0.3.0
Determining dependency 'glib-2.0' with pkg-config executable '/usr/bin/pkg-config'
Native dependency glib-2.0 found: YES 2.56.1
Determining dependency 'gio-unix-2.0' with pkg-config executable '/usr/bin/pkg-config'
Native dependency gio-unix-2.0 found: YES 2.56.1
Determining dependency 'gthread-2.0' with pkg-config executable '/usr/bin/pkg-config'
Native dependency gthread-2.0 found: YES 2.56.1
Determining dependency 'gmodule-no-export-2.0' with pkg-config executable '/usr/bin/pkg-config'
Native dependency gmodule-no-export-2.0 found: YES 2.56.1
Determining dependency 'gtk+-3.0' with pkg-config executable '/usr/bin/pkg-config'
Native dependency gtk+-3.0 found: YES 3.22.30
Determining dependency 'cairo' with pkg-config executable '/usr/bin/pkg-config'
Native dependency cairo found: YES 1.15.12
Running compile:
Working directory:  /tmp/tmpnpsnphqv
Command line:  ccache cc /tmp/tmpnpsnphqv/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -c -o /tmp/tmpnpsnphqv/output.obj -O0 -Wall 

Code:
 int i;

Compiler stdout:
 
Compiler stderr:
 
Compiler for C supports arguments -Wall: YES
Running compile:
Working directory:  /tmp/tmpyadbcv0l
Command line:  ccache cc /tmp/tmpyadbcv0l/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -c -o /tmp/tmpyadbcv0l/output.obj -O0 -Wextra 

Code:
 int i;

Compiler stdout:
 
Compiler stderr:
 
Compiler for C supports arguments -Wextra: YES
Running compile:
Working directory:  /tmp/tmpqzbb6msk
Command line:  ccache cc /tmp/tmpqzbb6msk/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -c -o /tmp/tmpqzbb6msk/output.obj -O0 -pedantic 

Code:
 int i;

Compiler stdout:
 
Compiler stderr:
 
Compiler for C supports arguments -pedantic: YES
Running compile:
Working directory:  /tmp/tmp2kusg42e
Command line:  ccache cc /tmp/tmp2kusg42e/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -c -o /tmp/tmp2kusg42e/output.obj -O0 -Werror=implicit-function-declaration 

Code:
 int i;

Compiler stdout:
 
Compiler stderr:
 
Compiler for C supports arguments -Werror=implicit-function-declaration: YES
Running compile:
Working directory:  /tmp/tmpc__trwgk
Command line:  ccache cc /tmp/tmpc__trwgk/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -c -o /tmp/tmpc__trwgk/output.obj -O0 -Werror=vla 

Code:
 int i;

Compiler stdout:
 
Compiler stderr:
 
Compiler for C supports arguments -Werror=vla: YES
Running compile:
Working directory:  /tmp/tmpry713019
Command line:  ccache cc /tmp/tmpry713019/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -c -o /tmp/tmpry713019/output.obj -O0 -fvisibility=hidden 

Code:
 int i;

Compiler stdout:
 
Compiler stderr:
 
Compiler for C supports arguments -fvisibility=hidden: YES
Determining dependency 'sqlite3' with pkg-config executable '/usr/bin/pkg-config'
Native dependency sqlite3 found: YES 3.22.0
Determining dependency 'synctex' with pkg-config executable '/usr/bin/pkg-config'
Dependency synctex found: NO
Running compile:
Working directory:  /tmp/tmppiirj0zf
Command line:  ccache cc /tmp/tmppiirj0zf/testfile.c -pipe -D_FILE_OFFSET_BITS=64 -o /tmp/tmppiirj0zf/output.exe -O0 -Wl,--start-group -lmagic -Wl,--end-group 

Code:
 int main(int argc, char **argv) { return 0; }
Compiler stdout:
 
Compiler stderr:
 
Library magic found: YES
Determining dependency 'libseccomp' with pkg-config executable '/usr/bin/pkg-config'
Dependency libseccomp found: NO
Configuring zathura-version.h using configuration
Determining dependency 'glib-2.0' with pkg-config executable '/usr/bin/pkg-config'
Native dependency glib-2.0 found: YES 2.56.1
Program desktop-file-validate found: YES (/usr/bin/desktop-file-validate)
Adding test "validate-desktop".
Program appstream-util found: YES (/usr/bin/appstream-util)
Adding test "validate-appdata".
Configuring zathura using configuration
Configuring _zathura using configuration
Program sphinx-build found: YES (/usr/bin/sphinx-build)
Determining dependency 'check' with pkg-config executable '/usr/bin/pkg-config'
Native dependency check found: YES 0.12.0
Adding test "document".
Adding test "session".
Adding test "utils".
Build targets in project: 13
Found ninja-1.8.2 at /usr/bin/ninja

```

**Comment by ePirat** [comment]
> `LIBRARY_PATH` does not mean anything

It does here, setting `LIBRARY_PATH` adds the path to the library search paths of the linker, which I guess affects cc.find_library, no?

**Comment by jpakkane** [comment]
Huh. I did not know that was a thing but apparently it is. So yes, it should work, but note that if you ever run a compile task with that envvar unset, it will fail to link (because the test we do is "does linking a dummy exe with `-lfoo` return success or not" and if it does, use that linker argument).

**Comment by ChukwuemezieChristopherAneke** [comment]
I see, thanks for clarifying. I can confirm I had the LIBRARY_PATH was not set until I manually set it. Never the less, I am closing this issue because I found an up to date package that negates the need to build from source. For future reference when using meson again, it would help to know apart from `.pc` files which `pkgconfig` looks for, what other file extension does the linker in meson look for? Is it `*.so` files or `*.so.1`
